apps/api/src/app/workflows-v2/usecases/create-variables-object/create-variables-object.usecase.ts

```diff
@@ -65,20 +66,56 @@ export class CreateVariablesObject {
       );
 
       const hasUsedEvents = !!(step.events && typeof step.events === 'string');
-      if (hasUsedEventCount || hasUsedEventsLength || hasUsedEvents) {
-        step.events = Array.isArray(step.events)
-          ? step.events
+      const { events } = step;
+      /**
+       * Check if events is an object and has a payload property.
+       */
+      const hasUsedEventsWithPayload = !!(
+        events &&
+        typeof events === 'object' &&
+        !Array.isArray(events) &&
+        'payload' in events
+      );
+      let variableNameAfterPayload = ['name'];
+      if (hasUsedEventsWithPayload) {
+        /**
+         * If events is an object and has a payload property, collect keys from the payload.
+         * e.g. [payload.foo.bar]
+         */
+        variableNameAfterPayload = collectKeys(events.payload);
+      }
+
+      if (hasUsedEventCount || hasUsedEventsLength || hasUsedEvents || hasUsedEventsWithPayload) {
```

Comment:
this case is not working: 
![Screenshot 2025-04-24 at 01 40 26](https://github.com/user-attachments/assets/3f37592b-ba5a-4796-b5ec-e2a4309bfc2b)